guides/2022-04-05-build-a-conduit-connector.mdx

```diff
@@ -0,0 +1,318 @@
+				Required:    true,
+				Description: "The Application ID for Algolia.",
+			},
+			DestinationConfigIndexName: {
+				Required:    true,
+				Description: "The Algolia index where records get written into.",
+			},
+		},
+		SourceParams: nil,
+	}
+}
+```
+
+## Entrypoint
+
+Next, let's look at `cmd/algolia/main.go`. This is our entrypoint. 
+
+```go
+package main
+
+import (
+	algolia "github.com/conduitio/conduit-connector-algolia"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+func main() {
+	sdk.Serve(
+		algolia.Specification,
+		nil,
+		algolia.NewDestination,
+		)
+}
+```
+
+`sdk.Serve` is TODO. Notice `nil` as the second parameter. This is due to this connector being just a Destination.
+
+## Destination
+
+The `destination.go` implements the connector lifecycle methods. 
+
+```go
+...
+
+type Destination struct {
+	sdk.UnimplementedDestination
+
+	config DestinationConfig
+	index  *search.Index
+}
+
+type DestinationConfig struct {
+	APIKey        string
+	ApplicationID string
+	IndexName     string
+}
+...
+```
+
+First we define two structs:
+
+- Destination - This is the "storage" struct of your connector.
+- DestinationConfig - This makes it easy to access connector config.
+
+### Configure
+
+`Configure` is always the first function to be called in a connector and is used to store and validate the connector configuration. 
+
+```go
+
+func (d *Destination) Configure(ctx context.Context, cfg map[string]string) error {
+	destCfg := DestinationConfig{
+		APIKey:        cfg[DestinationConfigAPIKey],
+		ApplicationID: cfg[DestinationConfigApplicationID],
+		IndexName:     cfg[DestinationConfigIndexName],
+	}
+
+	if destCfg.APIKey == "" {
+		return fmt.Errorf("%q is a required parameter", DestinationConfigAPIKey)
+	}
+	if destCfg.ApplicationID == "" {
+		return fmt.Errorf("%q is a required parameter", DestinationConfigApplicationID)
+	}
+	if destCfg.IndexName == "" {
+		return fmt.Errorf("%q is a required parameter", DestinationConfigIndexName)
+	}
+
+	d.config = destCfg
+	return nil
+}
+```
+
+### Open
+
+`Open` is called after Configure to signal the plugin it can prepare to
+start producing records. If needed, the plugin should open connections in
+this function. 
+
+```go
+
+func (d *Destination) Open(ctx context.Context) error {
+	client := search.NewClient(d.config.ApplicationID, d.config.APIKey)
+	index := client.InitIndex(d.config.IndexName)
+	d.index = index
+	return nil
+}
+```
+
+### Write
+
+`Write` receives a Record and is supposed to write the record to the
+destination right away.
+```go
+
+func (d *Destination) Write(ctx context.Context, record sdk.Record) error {
+	 , err := d.index.SaveObject(Object(record))
+	if err != nil {
+		return fmt.Errorf("could not save object: %w", err)
+	}
+
+	sdk.Logger(ctx).Debug().
+		Int("taskId", res.TaskID).
+		Str("objectId", res.ObjectID).
+		Msg("saved object")
+
+	return nil
+}
+
+```
+
+The `Object` methods serializes the record. This is implemented in `object.go`.
```

Comment:
`Object` is actually a type that changes the way the record is serialized to JSON. I think calling it a "method" is a bit misleading.